code/modules/networks/computer3/mainframe2/misc_terms.dm

```diff
@@ -735,303 +738,284 @@ TYPEINFO(/obj/machinery/networked/storage)
 
 		return
 
-	attackby(obj/item/W, mob/user)
-		if (istype(W, /obj/item/tank))
-			return attack_hand(user)
-		else if (isscrewingtool(W))
-			playsound(src.loc, 'sound/items/Screwdriver.ogg', 50, 1)
-			boutput(user, "You [src.locked ? "secure" : "unscrew"] the maintenance panel.")
-			src.panel_open = !src.panel_open
-			src.updateUsrDialog()
-			return
+	/// Handheld tanks and butts are both fair game here
+	proc/is_valid_tank(obj/item/I)
+		return (istype(I, /obj/item/tank) || istype(I, /obj/item/clothing/head/butt))
+
+	/// Determine if the simulator is ready, returns a list of a bool for whether it's ready, and a dialogue for the user
+	proc/can_simulate()
+		if(!(src.tank1 && src.tank2))
+			return list(FALSE, "Both tanks are required!")
+		else if (GET_COOLDOWN(global, "bomb_simulator"))
+			return list(FALSE, "Simulator not ready, please try again later.")
+		else if (src.vrbomb)
+			return list(TRUE, "Simulation in progress!")
 		else
-			..()
-		return
-
-	attack_hand(mob/user)
-		if(status & (NOPOWER|BROKEN))
-			return
-
-		if(user.lying || user.stat)
-			return 1
-
-		if ((BOUNDS_DIST(src, user) > 0 || !istype(src.loc, /turf)) && !issilicon(user))
-			return 1
-
-		src.add_dialog(user)
-
-		var/dat = "<html><head><title>SimUnit - \[[bank_id]]</title></head><body>"
-
-		dat += "<b>Tank One:</b> <a href='?src=\ref[src];tank=1'>[src.tank1 ? "Eject" : "None"]</a><br>"
-		dat += "<b>Tank Two:</b> <a href='?src=\ref[src];tank=2'>[src.tank2 ? "Eject" : "None"]</a><hr>"
-
-		dat += "<b>Simulation:</b> [vrbomb ? "IN PROGRESS" : "<a href='?src=\ref[src];simulate=1'>BEGIN</a>"]<br>"
-
-		var/readout_color = "#000000"
-		var/readout = "ERROR"
-		if(src.host_id)
-			readout_color = "#33FF00"
-			readout = "OK CONNECTION"
+			return list(TRUE, "Simulator ready.")
+
+#define TANK_ONE "1"
+#define TANK_TWO "2"
+/// Checks if we have a tank in the specified slot
+#define HAS_TANK(tanknum) ((tanknum == TANK_ONE) && src.tank1) || ((tanknum == TANK_TWO) && src.tank2)
+/// Add the tank to the slot being interact with in the device
+#define ADD_TANK(tanknum, tank) tank.set_loc(src); if ((tanknum) == TANK_ONE) {src.tank1 = tank;}\
+								else {src.tank2 = tank;};
+	/// Interact with one of the tank slots on the machine. "1" for the first slot, "2" for the second. "null" to use an empty slot, if there is one.
+	proc/interact_tank_slot(mob/user, obj/item/I, slot=null)
```

Comment:
Consider manually forcing a TGUI update here to make it a little more responsive.